gateway/ln-gateway/src/db.rs

```diff
@@ -646,4 +739,94 @@ mod fedimint_migration_tests {
         )
         .await
     }
+
+    #[tokio::test(flavor = "multi_thread")]
+    async fn test_isolated_db_migration() -> anyhow::Result<()> {
+        async fn create_isolated_record(prefix: Vec<u8>, db: &Database) {
+            // Create an isolated database the old way where there was no prefix
+            let isolated_db = db.with_prefix(prefix);
+            let mut isolated_dbtx = isolated_db.begin_transaction().await;
+
+            // Insert a record into the isolated db (doesn't matter what it is)
+            isolated_dbtx
+                .insert_new_entry(
+                    &GatewayPublicKey,
+                    &Keypair::new(secp256k1::SECP256K1, &mut rand::thread_rng()),
+                )
+                .await;
+            isolated_dbtx.commit_tx().await;
+        }
+
+        let federation_id = FederationId::from_str(
+            "0406afdc71a052d2787eab7e84c95803636d2a84c272eb81b4e01b27acb86c6f",
+        )
```

Comment:
Is `FederationId::dummy()` below meant to be a federation ID that _doesn't_ start with the clashing prefix? If so, can we declare that as a stringified ID like we do here, and name the two IDs something like `clashing_federation_id` and `non_clashing_federation_id`? Expecting `dummy()` to return a federation ID that doesn't start with this problematic prefix adds an implicit API contract that, while unlikely to be broken, would silently lose test coverage if it was.

Reply:
Done